lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -4208,7 +4212,7 @@ shelleyOnlyMkWithdrawal
     -> WalletId
     -> AnyCardanoEra
     -> ApiWithdrawalPostData
-    -> Handler Withdrawal
+    -> Handler (Withdrawal, Maybe (XPrv, Passphrase "encryption"))
```

Comment:
I am thinking of putting these optional credentials inside the `Withdrawal` data type:

```haskell
data Withdrawal
    = WithdrawalSelf RewardAccount (NonEmpty DerivationIndex) Coin
    | WithdrawalExternal 
        RewardAccount (NonEmpty DerivationIndex) Coin 
        (Maybe (XPrv, Passphrase "encryption")))
    | NoWithdrawal
    deriving (Show, Eq)
```
wdyt?

Reply:
I also thought about it when looking at this again yesterday 🤔 will give it a shot/some thought

Reply:
Regarding:

> ```hs
>     | WithdrawalExternal 
>        RewardAccount (NonEmpty DerivationIndex) Coin 
>        (Maybe (XPrv, Passphrase "encryption")))
> ```

For an external withdrawal, are there any circumstances for which passing `Nothing` for `(Maybe (XPrv, Passphrase "encryption"))` could result in a successful withdrawal?

Reply:
Done - I think it turned out pretty well 👍